node/tools/src/config.rs

```diff
@@ -48,19 +74,28 @@ impl ProtoFmt for NodeAddr {
 
 /// Node configuration including executor configuration, optional validator configuration,
 /// and application-specific settings (e.g. metrics scraping).
-#[derive(Debug, PartialEq)]
+#[derive(Debug, PartialEq, Clone)]
 pub struct AppConfig {
-    pub server_addr: std::net::SocketAddr,
-    pub public_addr: std::net::SocketAddr,
-    pub metrics_server_addr: Option<std::net::SocketAddr>,
+    pub server_addr: SocketAddr,
+    pub public_addr: SocketAddr,
+    pub metrics_server_addr: Option<SocketAddr>,
 
     pub validators: validator::ValidatorSet,
     pub genesis_block: validator::FinalBlock,
     pub max_payload_size: usize,
 
     pub gossip_dynamic_inbound_limit: usize,
     pub gossip_static_inbound: HashSet<node::PublicKey>,
-    pub gossip_static_outbound: HashMap<node::PublicKey, std::net::SocketAddr>,
+    pub gossip_static_outbound: HashMap<node::PublicKey, SocketAddr>,
+}
+
+impl AppConfig {
```

Comment:
Any reason for having `impl AppConfig` clause twice?

Reply:
oops, no reason, I missed it